### PR TITLE
Point2dVector2d: improved code quality

### PR DESCRIPTION
**Pull request by saeeedtorabi**
- fixed a bug for bisect of 2 unit vectors
- improved src code formatting
- improved documentation
- improved tests
- Improved Point2dArrayCarrier and Point3dArrayCarrier

### CODE REVIEW

File: common/api/core-geometry.api.md [hunk 1]
```diff
@@ -6095,12 +6096,14 @@ export type XAndY = Readonly<WritableXAndY>;
...
     protected constructor(x?: number, y?: number);
+    at(index: number): number;
     static crossProductToPoints(origin: XAndY, targetA: XAndY, targetB: XAndY): number;
     distance(other: XAndY): number;
     distanceSquared(other: XAndY): number;
     freeze(): Readonly<this>;
+    indexOfMaxAbs(): number;
     isAlmostEqual(other: XAndY, tol?: number): boolean;
-    isAlmostEqualMetric(other: XAndY): boolean;
+    isAlmostEqualMetric(other: XAndY, radianTol?: number): boolean;
```
Comment (dassaf4) [inline common/api/core-geometry.api.md:6106]: Should be `distanceTol`, as in a tolerance in meters.

File: core/geometry/src/geometry3d/Point2dVector2d.ts [hunk 2]
```diff
@@ -139,142 +196,168 @@ export class Point2d extends XY implements BeJSONFunctions {
...
-    return (targetA.x - this.x) * (targetB.x - this.x) +
-      (targetA.y - this.y) * (targetB.y - this.y);
+    return (targetA.x - this.x) * (targetB.x - this.x) + (targetA.y - this.y) * (targetB.y - this.y);
   }
-
-  /** Returns the (scalar) cross product of two points/vectors, computed from origin to target1 and target2 */
+  /**
+   * Returns the (scalar) cross product of vector from this to targetA and vector from this to targetB
+   * @param targetA target of first vector
+   * @param targetB target of second vector
```
Comment (dassaf4) [inline core/geometry/src/geometry3d/Point2dVector2d.ts:326]: targetA and targetB don't agree with the names of the arguments

Comment (saeeedtorabi) [reply core/geometry/src/geometry3d/Point2dVector2d.ts:326]: will fix

File: common/api/core-geometry.api.md (hunk 1 above)
Comment (saeeedtorabi) [reply common/api/core-geometry.api.md:6106]: will fix

**Review by dassaf4 — APPROVED** [review]

**Review by bbastings — APPROVED** [review]

**Pull request merged**